### PR TITLE
npm guide

### PR DESCRIPTION
**Pull request by ascott1**
This adds a really barebones guide to publishing code on npm.

## Review

- @wpears 
- @contolini

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,18 @@
 All notable changes to this project will be documented in this file.
 We follow the [Semantic Versioning 2.0.0](http://semver.org/) format.
 
+## 2015-04-08
```
Comment (jimmynotjim) [inline CHANGELOG.md]: There's a note above about semver but none of these changes include release numbers. Should that note be removed?

Comment (ascott1) [inline CHANGELOG.md]: Yeah, I *think* so. I like the idea of maintain a changelog, but versioning this seems hard.

**Comment by wpears** [comment]
I like [substack](http://substack.net/how_I_write_modules) on the general module writing process. One of node's greatest strengths is its vibrant ecosystem, so checking around for the module you're about to write is always a good first step.

**Comment by ascott1** [comment]
Good call @contolini.

@wpears I'm going to add a link to Substack's guide as I agree it's worth a read (and that searching for existing modules is a good first step). Feel free to add or tweak the wording of this document from there.

**Comment by ascott1** [comment]
Updated ^^

**Comment by wpears** [comment]
Looks good; merging.

**Pull request merged**